dask/array/overlap.py

```diff
@@ -15,6 +15,25 @@
 from ..utils import concrete
 
 
+class Depth:
+    def __init__(self, axes):
+        self.one = [0] * len(axes)
+        self.two = [0] * len(axes)
+        self.total = [0] * len(axes)
+
+        for i,idepth in axes.items():
+            idepth = axes.get(i,0)
+            if isinstance(idepth,tuple):
+                self.one[i] = idepth[0]
+                self.two[i] = idepth[1]
+                self.total[i] = self.one[i] + self.two[i]
+
+            else:
+                self.one[i] = idepth
+                self.two[i] = idepth
+                self.total[i] = self.one[i] + self.two[i]
```

Comment:
Would it be possible to adjust this change to work with out this `Depth` class? I'm hopeful we can achieve the same end just using `tuple`s with different depths (similar to how it was before).

Reply:
yes.  That said, while the current class implementation is less than perfect, I do believe it trends toward improved readability.

Reply:
I'm not sure about that. This looks like it will increase the maintenance burden here, which would be nice to avoid.

Reply:
I certainly understand that factor.  I will revisit the current (preliminary) implementation with special consideration to maintainability.

Reply:
It looks like this is the major blocker.  @mpeaton I think that the ask here is that you switch back to using tuples.  In general in this project we try to keep to core data structures when possible.  This ends up keeping things somewhat easier to review, especially because people jump around a lot, and tend not to invest much in abstractions for any particular part of the code.